WalletWasabi.Fluent/ViewModels/CoinControl/Core/PocketCoinControlItemViewModel.cs

```diff
@@ -1,10 +1,16 @@
 using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using DynamicData;
+using ReactiveUI;
 using WalletWasabi.Fluent.Models;
 
 namespace WalletWasabi.Fluent.ViewModels.CoinControl.Core;
 
-public class PocketCoinControlItemViewModel : CoinControlItemViewModelBase
+public class PocketCoinControlItemViewModel : CoinControlItemViewModelBase, IDisposable
```

Comment:
I am wondering if the `IDisposable` is really needed?

In the case of `IsSelected` surely not, it subscribes to its own item.
In the case of `Children` I am not sure... It is still its own item, but `.AsObservableChangeSet()` or the `BindTo` changes something? It would be worth an investigation.

That would simplify a lot if we wouldn't have to deal with the disposing.

Reply:
I did the investigation:

![image](https://user-images.githubusercontent.com/3109851/206879884-e6beeb5e-1866-4b04-9b00-deded8a328ef.png)

```csharp
    public class UnitTest1
    {
        public UnitTest1(ITestOutputHelper output)
        {
            DotMemoryUnitTestOutput.SetOutputMethod(output.WriteLine);
        }

        [Fact]
        public void No_disposal()
        {
            var children = new[] { new Child(), new Child() };
            var sut = new Parent(children);
            sut = null;
            GC.Collect();
            dotMemory.Check(memory => memory.GetObjects(where => where.Type.Is<Parent>()).ObjectsCount.Should().Be(0));
        }

        [Fact]
        public void Proper_disposal()
        {
            var children = new[] { new Child(), new Child() };
            var sut = new ParentWithProperDisposal(children);
            sut.Dispose();
            sut = null;
            dotMemory.Check(memory => memory.GetObjects(where => where.Type.Is<Parent>()).ObjectsCount.Should().Be(0));

        }
    }

    public class Parent
    {
        public Parent(Child[] children)
        {
            children.AsObservableChangeSet()
                .WhenPropertyChanged(x => x.IsSelected)
                .Select(x => true)
                .BindTo(this, x => x.IsSelected);
        }

        [Reactive]
        public bool IsSelected { get; set; }
    }

    public class ParentWithProperDisposal : IDisposable
    {
        private readonly CompositeDisposable disposables = new();

        public ParentWithProperDisposal(Child[] children)
        {
            children.AsObservableChangeSet()
                .WhenPropertyChanged(x => x.IsSelected)
                .Select(x => true)
                .BindTo(this, x => x.IsSelected)
                .DisposeWith(disposables);
        }

        [Reactive]
        public bool IsSelected { get; set; }

        public void Dispose()
        {
            disposables.Dispose();
        }
    }

    public class Child : ReactiveObject
    {
        [Reactive]
        public bool IsSelected { get; set; }
    }
```

This shows 2 tests: One doesn't dispose the subscription. The other does.
1. When the subscription isn't disposed, the parent object is never released as long as the children are alive.
2. When the subscription is disposed, the object is destroyed properly.

The rule of thumb is: Whenever a subscription is made (IDisposable is returned) and you are observing anything whose lifecycle isn't tied to the current object, it needs to be disposed explicitly.